torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -15,6 +15,21 @@ namespace c10d {
 
 namespace {
 
+struct AutoNcclGroup {
```

Comment:
Question, is this the same as the one in `nccl.h` except the error checking? Is it possible to directly use that one here?

Reply:
Yes, the code here is identical to the one in `nccl.h` except the error checking macro.  This 'new' `AutoNcclGroup` is local to the `ProcessGroupNCCL.cpp` file.  I chose to repeat the code here instead of trying to add the header dependency to `torch/csrc/cuda` when the code there is depending on c10 -- circular dep?  I'm not yet completely familiar with how the PyTorch directories and libraries are structured.  Let me know how I should proceed.  Thanks.

Reply:
I see, thanks! Then, let's keep it as is.